libV2/schemaUtils.js

```diff
@@ -0,0 +1,1620 @@
+
+    return resolvedExample;
+  },
+
+  resolveExampleData = (context, exampleData) => {
+    if (_.has(exampleData, '$ref')) {
+      const resolvedRef = resolveRefForExamples(context, exampleData.$ref);
+      exampleData = resolveExampleData(context, resolvedRef);
+    }
+    else if (typeof exampleData === 'object') {
+      _.forOwn(exampleData, (data, key) => {
+        exampleData[key] = resolveExampleData(context, data);
+      });
+    }
+
+    return exampleData;
+  },
+
+  /**
+   * returns first example in the input map
+   * @param {Object} context - Global context object
+   * @param {Object} exampleObj - Object defined in the schema
+   * @returns {*} first example in the input map type
+   */
+  getExampleData = (context, exampleObj) => {
+    let example = {},
+      exampleKey;
+
+    if (typeof exampleObj !== 'object') {
+      return '';
+    }
+
+    exampleKey = Object.keys(exampleObj)[0];
+    example = exampleObj[exampleKey];
+
+    if (example.$ref) {
+      example = resolveExampleData(context, example);
+    }
+
+    if (_.get(example, 'value')) {
+      example = resolveExampleData(context, example.value);
+    }
+
+    return example;
+  },
+
+  /**
+   * Handle resoltion of allOf property of schema
+   *
+   * @param {Object} context - Global context object
+   * @param {Object} schema - Schema to be resolved
+   * @param {Number} [stack] - Current recursion depth
+   * @param {*} resolveFor - resolve refs for flow validation/conversion (value to be one of VALIDATION/CONVERSION)
+   * @param {Object} seenRef - Map of all the references that have been resolved
+   *
+   * @returns {Object} Resolved schema
+   */
+  resolveAllOfSchema = (context, schema, stack, resolveFor = CONVERSION, seenRef = {}) => {
+    try {
+      return mergeAllOf(_.assign(schema, {
+        allOf: _.map(schema.allOf, (schema) => {
+          // eslint-disable-next-line no-use-before-define
+          return resolveSchema(context, schema, stack, resolveFor, _.cloneDeep(seenRef));
+        })
+      }), {
+        resolvers: {
+          // for keywords in OpenAPI schema that are not standard defined JSON schema keywords, use default resolver
+          defaultResolver: (compacted) => { return compacted[0]; }
+        }
+      });
+    }
+    catch (e) {
+      console.warn('Error while resolving allOf schema: ', e);
+      return { value: '<Error: Could not resolve allOf schema' };
+    }
+  },
+
+  /**
+   * Resolve a given ref from the schema
+   *
+   * @param {Object} context - Global context
+   * @param {Object} schema - Schema that is to be resolved
+   * @param {Number} [stack] - Current recursion depth
+   * @param {String} resolveFor - For which action this resoltion is to be done
+   * @param {Object} seenRef - Map of all the references that have been resolved
+   *
+   * @returns {Object} Returns the object that staisfies the schema
+   */
+  resolveSchema = (context, schema, stack = 0, resolveFor = CONVERSION, seenRef = {}) => {
```

Comment:
this seen ref should ideally be a directed graph/tree. Leave a comment here that this needs to be refactored.